src/test/kotlin/io/meshcloud/dockerosb/metrics/GaugeMetricsServiceTest.kt

```diff
@@ -0,0 +1,61 @@
+    Assert.assertEquals(serviceInstanceId,test1?.serviceInstanceId)
+    Assert.assertEquals(resource1, test1?.resource)
+    Assert.assertEquals(10, test1?.values?.count())
+
+    // can read multiple files for the same serviceInstanceId and time filter works
+    val test2 = gaugeController.getGaugeMetricValues(serviceDefinitionId, secondStartDate, endDate).body?.dataPoints?.find { serviceInstanceDatapoints -> (serviceInstanceDatapoints.serviceInstanceId == serviceInstanceId && serviceInstanceDatapoints.resource==resource1 ) }
+    Assert.assertEquals(serviceInstanceId,test2?.serviceInstanceId)
+    Assert.assertEquals(resource1, test2?.resource)
+    Assert.assertEquals(6, test2?.values?.count())
+
+    // can separate other serviceInstanceIds
+    val test3 = gaugeController.getGaugeMetricValues(serviceDefinitionId, firstStartDate, endDate).body?.dataPoints?.find { serviceInstanceDatapoints -> (serviceInstanceDatapoints.serviceInstanceId == serviceInstanceId2 && serviceInstanceDatapoints.resource==resource1 ) }
+    Assert.assertEquals(serviceInstanceId2,test3?.serviceInstanceId)
+    Assert.assertEquals(resource1, test3?.resource)
+    Assert.assertEquals(5, test3?.values?.count())
+
+    // can separate other serviceInstanceIds and time filter works
+    val test4 = gaugeController.getGaugeMetricValues(serviceDefinitionId, secondStartDate, endDate).body?.dataPoints?.find { serviceInstanceDatapoints -> (serviceInstanceDatapoints.serviceInstanceId == serviceInstanceId2 && serviceInstanceDatapoints.resource==resource1 ) }
+    Assert.assertEquals(serviceInstanceId2,test4?.serviceInstanceId)
+    Assert.assertEquals(resource1, test4?.resource)
+    Assert.assertEquals(3, test4?.values?.count())
+
+    // can separate different resources for the serviceInstanceId
+    val test5 = gaugeController.getGaugeMetricValues(serviceDefinitionId, firstStartDate, endDate).body?.dataPoints?.find { serviceInstanceDatapoints -> (serviceInstanceDatapoints.serviceInstanceId == serviceInstanceId && serviceInstanceDatapoints.resource==resource2 ) }
+    Assert.assertEquals(serviceInstanceId,test5?.serviceInstanceId)
+    Assert.assertEquals(resource2, test5?.resource)
+    Assert.assertEquals(6, test5?.values?.count())
+
+    // can separate different resources for the serviceInstanceId and time filter works
+    val test6 = gaugeController.getGaugeMetricValues(serviceDefinitionId, secondStartDate, endDate).body?.dataPoints?.find { serviceInstanceDatapoints -> (serviceInstanceDatapoints.serviceInstanceId == serviceInstanceId && serviceInstanceDatapoints.resource==resource2 ) }
+    Assert.assertEquals(serviceInstanceId,test6?.serviceInstanceId)
+    Assert.assertEquals(resource2, test6?.resource)
+    Assert.assertEquals(4, test6?.values?.count())
```

Comment:
s: Separating out the test cases into one function per case would allow to check specific behavior by running just this test and ease development in the future.

Same for the other new tests.